test/builder_test.c

```diff
@@ -66,6 +66,69 @@ TEST test_builder_write_uint8_split_packet()
     PASS();
 }
 
+TEST test_builder_write_uint8_exceeds_small_max()
+{
+    trilogy_buffer_t buff;
+    int err = trilogy_buffer_init(&buff, 1);
+    ASSERT_OK(err);
+
+    trilogy_builder_t builder;
+    err = trilogy_builder_init(&builder, &buff, 0);
+    ASSERT_OK(err);
+
+    // It's called "max", but it's really the length that will fail;
+    // this matches the server's behavior.
+    err = trilogy_builder_set_max_packet_length(&builder, 3);
+    ASSERT_OK(err);
+
+    err = trilogy_builder_write_uint8(&builder, 0x01);
+    ASSERT_OK(err);
+
+    err = trilogy_builder_write_uint8(&builder, 0x02);
+    ASSERT_OK(err);
+
+    err = trilogy_builder_write_uint8(&builder, 0x03);
+    ASSERT_EQ(TRILOGY_MAX_PACKET_EXCEEDED, err);
+
+    trilogy_buffer_free(&buff);
+    PASS();
+}
+
+TEST test_builder_write_uint8_exceeds_large_max()
+{
+    trilogy_buffer_t buff;
+    int err = trilogy_buffer_init(&buff, 1);
+    ASSERT_OK(err);
+
+    trilogy_builder_t builder;
+    err = trilogy_builder_init(&builder, &buff, 0);
+    ASSERT_OK(err);
+
+    size_t max = TRILOGY_MAX_PACKET_LEN * 2;
+    err = trilogy_builder_set_max_packet_length(&builder, max);
+    ASSERT_OK(err);
+
+    size_t len = max - 3;
+    uint8_t *bytes = malloc(len);
```

Comment:
Can we use `calloc` or similar to avoid reading uninitialized values. It doesn't really matter since this is a test, but fixing that should allow us to run under valgrind or sanitizers.

Reply:
Done 👍🏻 (including the existing ones)